### ISSUE TITLE
v0.8.0 "Illegal instruction" error

### ISSUE THREAD

**Issue by FengLinqing** [ospray/OSPRay#16]
I got a "Illegal instruction: 4" error when I was trying to run the ospVolumeViewer. 
If I change cmake option "OSPRAY_BUILD_ISA" to "SSE" then it works fine. Is this a bug of the new runtime ISA selection function? I am using OSX 10.9.5 and my CPU does not have AVX support.

**Comment by ingowald** [comment]
Feng,

Thanks for reporting this; we will be looking into this ASAP.

We have indeed added this multi-isa feature only a few days ago, and 
though we did test it with running a "ALL" build on both AVX2 and 
AVX1-only machines we did not test it with an SSE-only machine (they are 
getting kind of rare these days). It most certainly sounds like a 
ISA-selection bug; we'll be looking into it right away.

Cheers,

Ingo

**Comment by gregjohnson** [comment]
Hi Feng,

We're attempting to reproduce this. Can you give us some additional information?

- the output of

    sysctl -a | grep machdep.cpu

- which compiler / version did you use?

- can you run the volume viewer again, but with the "--osp:debug" command line option, and give us the output?

Thanks,
Greg

**Comment by FengLinqing** [comment]
Hi Greg,

1. output:
machdep.cpu.max_basic: 11
machdep.cpu.max_ext: 2147483656
machdep.cpu.vendor: GenuineIntel
machdep.cpu.brand_string: Intel(R) Xeon(R) CPU           W3530  @ 2.80GHz
machdep.cpu.family: 6
machdep.cpu.model: 26
machdep.cpu.extmodel: 1
machdep.cpu.extfamily: 0
machdep.cpu.stepping: 5
machdep.cpu.feature_bits: 3219913727 10281917
machdep.cpu.extfeature_bits: 672139520 1
machdep.cpu.signature: 67237
machdep.cpu.brand: 0
machdep.cpu.features: FPU VME DE PSE TSC MSR PAE MCE CX8 APIC SEP MTRR PGE MCA CMOV PAT PSE36 CLFSH DS ACPI MMX FXSR SSE SSE2 SS HTT TM PBE SSE3 DTES64 MON DSCPL VMX EST TM2 SSSE3 CX16 TPR PDCM SSE4.1 SSE4.2 POPCNT
machdep.cpu.extfeatures: SYSCALL XD EM64T LAHF RDTSCP TSCI
machdep.cpu.logical_per_package: 16
machdep.cpu.cores_per_package: 8
machdep.cpu.microcode_version: 17
machdep.cpu.processor_flag: 1
machdep.cpu.mwait.linesize_min: 64
machdep.cpu.mwait.linesize_max: 64
machdep.cpu.mwait.extensions: 3
machdep.cpu.mwait.sub_Cstates: 4384
machdep.cpu.thermal.sensor: 1
machdep.cpu.thermal.dynamic_acceleration: 1
machdep.cpu.thermal.invariant_APIC_timer: 0
machdep.cpu.thermal.thresholds: 2
machdep.cpu.thermal.ACNT_MCNT: 1
machdep.cpu.thermal.core_power_limits: 0
machdep.cpu.thermal.fine_grain_clock_mod: 0
machdep.cpu.thermal.package_thermal_intr: 0
machdep.cpu.thermal.hardware_feedback: 0
machdep.cpu.thermal.energy_policy: 0
machdep.cpu.arch_perf.version: 3
machdep.cpu.arch_perf.number: 4
machdep.cpu.arch_perf.width: 48
machdep.cpu.arch_perf.events_number: 7
machdep.cpu.arch_perf.events: 68
machdep.cpu.arch_perf.fixed_number: 3
machdep.cpu.arch_perf.fixed_width: 48
machdep.cpu.cache.linesize: 64
machdep.cpu.cache.L2_associativity: 8
machdep.cpu.cache.size: 256
machdep.cpu.tlb.inst.small: 64
machdep.cpu.tlb.inst.large: 7
machdep.cpu.tlb.data.small: 64
machdep.cpu.tlb.data.large: 32
machdep.cpu.tlb.shared: 512
machdep.cpu.address_bits.physical: 36
machdep.cpu.address_bits.virtual: 48
machdep.cpu.core_count: 4
machdep.cpu.thread_count: 8
machdep.cpu.flex_ratio.desired: 63
machdep.cpu.flex_ratio.min: 12
machdep.cpu.flex_ratio.max: 21

2. I am using Xcode 6.2 and output of clang++ --version:
Apple LLVM version 6.0 (clang-600.0.57) (based on LLVM 3.5svn)
Target: x86_64-apple-darwin13.4.0
Thread model: posix

3. It seems that the error only occurs in release mode. The debug version can start and render scene correctly. When I tried to run release version of "./ospVolumeViewer --osp:debug" I got "Illegal instruction: 4". This is the output from debug version of ospVolumeViewer:

$ ./ospVolumeViewer --osp:debug
Embree Ray Tracing Kernels 2.3.3 (Apr 25 2015)
  Compiler : CLANG 6.0 (clang-600.0.57)
  Platform : Mac OS X (64bit)
  CPU      : SSE SSE2 SSE3 SSSE3 SSE41 SSE42 POPCNT 
  Features : intersection_filter bufferstride 
general:
  build threads = 1
  verbosity     = 2
triangles:
  accel         = default
  builder       = default
  traverser     = default
  replications  = 2
motion blur triangles:
  accel         = default
  builder       = default
  traverser     = default
hair:
  accel         = default
  builder       = default
  traverser     = default
  replications  = 3
subdivision surfaces:
  accel         = default
 
  USAGE:  ./ospVolumeViewer <filename> [filename] ... [options]
 
  Options:
 
    -benchmark <warm-up frames> <frames> : run benchmark and report overall frame rate
    -dt <dt>                             : use ray cast sample step size 'dt'
    -module <moduleName>                 : load the module 'moduleName'
    -ply <filename>                      : load PLY geometry from 'filename'
    -rotate <rate>                       : automatically rotate view according to 'rate'
    -showframerate                       : show the frame rate in the window title bar
    -slice <filename>                    : load volume slice from 'filename'
    -transferfunction <filename>         : load transfer function from 'filename'
    -viewsize <width>x<height>           : force OSPRay view size to 'width'x'height'
    -viewup <x> <y> <z>                  : set viewport up vector to ('x', 'y', 'z')
    -writeframes <filename>              : emit frames to 'filename_xxxxx.ppm'

Thanks,
Feng
